src/compass/utils/lut.py

```diff
@@ -12,242 +12,150 @@
+
+    # Check which corrections are requested and accumulate corresponding data
+    # Geometrical and steering doppler
+    if lut_par.geometry_steering_doppler:
+        doppler = burst.doppler_induced_range_shift(range_step=rg_step,
+                                                    az_step=az_step)
+        doppler_meter = doppler.data * isce3.core.speed_of_light * 0.5
+        rg_data += doppler_meter
+        data_dict['doppler'][0] = doppler_meter
+
+    # Bistatic delay
+    if lut_par.bistatic_delay:
+        bistatic_delay = burst.bistatic_delay(range_step=rg_step,
+                                              az_step=az_step).data
+        az_data -= bistatic_delay
+        data_dict['bistatic_delay'][0] = -bistatic_delay
+
+    # Azimuth FM-rate mismatch
+    if lut_par.azimuth_fm_rate:
+        az_fm_rate = burst.az_fm_rate_mismatch_from_llh(lat, lon, height,
                                                         ellipsoid,
                                                         burst.as_isce3_radargrid(
                                                             az_step=az_step,
-                                                            rg_step=rg_step)
-                                                        )
-
-    # compute Solid Earth Tides using pySolid. Decimate the rdr2geo layers.
-    # compute decimation factor assuming a 5 km spacing along slant range
-    dec_factor = int(np.round(5000.0 / rg_step))
-    dec_slice = np.s_[::dec_factor]
-    rg_set_temp, az_set_temp = solid_earth_tides(burst, lat[dec_slice],
-                                                 lon[dec_slice],
-                                                 inc_angle[dec_slice],
-                                                 head_angle[dec_slice])
-
-    # Resize SET to the size of the correction grid
-    out_shape = bistatic_delay.data.shape
-    kwargs = dict(order=1, mode='edge', anti_aliasing=True,
-                  preserve_range=True)
-    rg_set = resize(rg_set_temp, out_shape, **kwargs)
-    az_set = resize(az_set_temp, out_shape, **kwargs)
-
-    # Compute ionosphere delay
-    los_ionosphere = ionosphere_delay(burst.sensing_mid,
-                                      burst.wavelength,
-                                      tec_path, lon, lat, inc_angle)
-
-    # Compute wet and dry troposphere delays using RAiDER
-    wet_los_tropo, dry_los_tropo, los_static_tropo =\
-        [np.zeros(out_shape) for _ in range(3)]
-
-    if weather_model_path is None:
-        # Compute static troposphere correction
+                                                            rg_step=rg_step)).data
+        az_data -= az_fm_rate
+        data_dict['azimuth_fm_rate'][0] = -az_fm_rate
+
+    # Solid Earth tides
+    if lut_par.solid_earth_tides:
+        dec_factor = int(np.round(5000.0 / rg_step))
+        dec_slice = np.s_[::dec_factor]
+        rg_set_temp, az_set_temp = solid_earth_tides(burst, lat[dec_slice],
```

Comment:
Can you take a look at `lat[dec_slice]` and others to see if its shape is as you intended?  FYI, please take a look at the comment in the link below to see what I've found in SET azimuth PR.

https://github.com/opera-adt/COMPASS/pull/105#discussion_r1136531890

Reply:
Thank you, I think this has been fixed in the latest commit